app/controllers/api/v2/concerns/gbv_key_performance_indicators.rb

```diff
@@ -0,0 +1,487 @@
+# frozen_string_literal: true
+
+module Api::V2::Concerns
```

Comment:
Following a similar sort of route as the Indicators (sort of) here. I'm not happy with the architecture of this right now given that the underlying search technology is in flux, but it was the natural, OOP way to meet the Rubocop requirements.

Reply:
I like that you divided it up into classes. The rule is: if you are finding yourself creating a new class, unless it's a super-specific thing, it deserves it's own file. 

This file has mostly business logic in it, and really is not a controller concern (an aide into taking input from the client and routing it to the right model or service with some light formatting) at all. We can review this together but I suspect that 1. you can extract this into multiple, testable classes, and 2. you can consolidate this with the KPI approach on the comment below.

Reply:
I also suspect that there is generic code in here that can be extracted into Service objects: essentially non-categorical POROs that take an input, perform some business or boilerplate logic and provide an output